src/CMakeLists.txt

```diff
@@ -255,19 +255,26 @@ ELSE()
 ENDIF()
 
 # Optional external dependency: zlib
-FIND_PACKAGE(ZLIB)
-IF (ZLIB_FOUND)
-	LIST(APPEND LIBGIT2_INCLUDES ${ZLIB_INCLUDE_DIRS})
-	LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
-	IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-		LIST(APPEND LIBGIT2_LIBS "z")
-		LIST(APPEND LIBGIT2_PC_LIBS "-lz")
+IF (NOT USE_BUNDLED_ZLIB)
+	FIND_PACKAGE(ZLIB)
+	IF (NOT USE_BUNDLED_ZLIP AND ZLIB_FOUND)
+		LIST(APPEND LIBGIT2_INCLUDES ${ZLIB_INCLUDE_DIRS})
+		LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
+		IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+			LIST(APPEND LIBGIT2_LIBS "z")
+			LIST(APPEND LIBGIT2_PC_LIBS "-lz")
+		ELSE()
+			SET(LIBGIT2_PC_REQUIRES "${LIBGIT2_PC_REQUIRES} zlib")
+		ENDIF()
+		ADD_FEATURE_INFO(zlib ON "Zlib support")
+	ENDIF()
+ENDIF()
+IF (USE_BUNDLED_ZLIB OR NOT ZLIB_FOUND)
+	IF (USE_BUNDLED_ZLIB)
+	    MESSAGE(STATUS "zlib: user opted to use bundled 3rd-party sources." )
```

Comment:
I don't think this message is required. The user will probably know when he's opted in to that